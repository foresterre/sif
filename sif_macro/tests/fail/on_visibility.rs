// see also 'tests/ok/visibility_2'

#[macro_use]
extern crate sif_macro;

pub mod a {
    #[parameterized]
    #[case(Some(0))]
    #[case(None)]
    pub(in crate::b) fn my_test(v: Option<i32>) {}
}

mod b {
    #[cfg(test)]
    fn call() {
        a::my_test::case_0(); // this is ok
    }
}

fn main() {
    if cfg!(test) {
        a::my_test::case_0(); // this isn't ok
    }
}
