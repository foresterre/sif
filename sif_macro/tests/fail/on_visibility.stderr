error[E0433]: failed to resolve: could not find `my_test` in `a`
  --> $DIR/on_visibility.rs:22:12
   |
22 |         a::my_test::case_0(); // this isn't ok
   |            ^^^^^^^ could not find `my_test` in `a`
